browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -189,7 +189,7 @@ EngineFlags ShouldBlockRequestOnTaskRunner(
       url_to_check, ctx->resource_type, source_host,
       ctx->aggressive_blocking || force_aggressive,
       &previous_result.did_match_rule, &previous_result.did_match_exception,
-      &previous_result.did_match_important, &ctx->mock_data_url);
+      &previous_result.did_match_important, &ctx->adblock_data_or_url);
```

Comment:
this is no longer relevant, right? If so can we change it back so it's easier to focus on the actual changes?

Reply:
I think it's definitely confusing and probably not correct that `ShouldMockRequest` returns true for this?

Reply:
> How do we differentiate between a url that is for a stub response and one that is for a redirect?

There isn't really any differentiation; this is getting populated either with a `data:` URL or a `http[s]://` URL. I was expecting to see a flag of some sort, but I guess this works? I would like to see some comment at least.

> I think it's definitely confusing and probably not correct that ShouldMockRequest returns true for this?

_Technically_ we are still mocking original files from websites with this mechanism, but the mocked response needs to be fetched externally first. Maybe a better name for that method would be something like `HasAdblockRedirection`.

Reply:
> There isn't really any differentiation; this is getting populated either with a data: URL or a http[s]:// URL. I was expecting to see a flag of some sort, but I guess this works? I would like to see some comment at least.

@antonok-edm my initial thought was to add a flag to context, but I saw Ivan's comment somewhere about how we're passing around a lot of information that can be derived in the context, so decided to do this (because data vs https URLs can be easily differentiated). Let me know if you don't agree. I can definitely add a comment to BraveRequestInfo.

> I think it's definitely confusing and probably not correct that ShouldMockRequest returns true for this?

I had changed this to `ShouldMockOrRedirectRequest`, but changed it back - that or `HasAdblockRedirection` both sound good to me.